### PR TITLE
rpc: Remove deprecated -rpcserialversion

### PR DESCRIPTION
**Pull request by maflcko**
The flag is problematic for many reasons:

* It is deprecated
* It is a global flag, requiring a restart to change, as opposed to a flag that can be set on each RPC invocation
* It may be hidden in config files by accident, hard to debug, causing LND crashes and bugs, see https://github.com/bitcoin/bitcoin/issues/28730#issuecomment-1780940868
* It makes performance improvements harder to implement: https://github.com/bitcoin/bitcoin/pull/17529#issuecomment-556082818

Fix all issues by removing it.

If there is a use-case, likely a per-RPC flag can be added, if needed.

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK.

**Comment by ajtowns** [comment]
> It is deprecated

It's only been deprecated in master for a couple of months; better to wait for it to actually be deprecated in the current release for a little while to give people a chance to complain if they're still relying on this functionality?

Otherwise, changes here look like what I'd expect.

> If there is a use-case, likely a per-RPC flag can be added, if needed.

Could give bitcoin-util the capability to strip witness data from a block/tx and leave it up to users who need it, rather than making the RPC API more complex for what's presumably a very rare use case.

**Comment by kristapsk** [comment]
Concept ACK

**Comment by ajtowns** [comment]
> The deprecation was covered in https://bitcoinops.org/en/newsletters/2023/09/20/ and 26.0 was released a few weeks ago. Unless anyone heard someone complain, this seems good to move forward now?

Given we're doing a short cycle, probably should either merge this soon for 27.0 or defer it to 28.0 and merge it shortly after branch off. No opinion either way on that personally.

crACK fa46cc22bc696e6845915ae91d6b68e36bf4c242

**Review by sedited — APPROVED** [review]
lgtm ACK fa46cc22bc696e6845915ae91d6b68e36bf4c242

> If there is a use-case, likely a per-RPC flag can be added, if needed.

**Comment by theuni** [comment]
Post-merge utACK fa46cc22bc696e6845915ae91d6b68e36bf4c242. Nice cleanup :)

**Pull request merged**
